test-src/api_test.js

```diff
@@ -0,0 +1,174 @@
+let workerType = require('../lib/worker-type');
+let workerState = require('../lib/worker-state');
+var assume = require('assume');
+var helper = require('./helper');
+var slugid = require('slugid');
+var assume = require('assume');
+var taskcluster = require('taskcluster-client');
+var _ = require('lodash');
+
+describe('worker-type API methods', () => {
+  let WorkerType, WorkerState, wt;
+
+  let cleanTables = async () => {
+    // remove all rows from either table
+    await WorkerType.scan({} , {
+      handler: async (item) => { await item.remove();  },
+    });
+    await WorkerState.scan({} , {
+      handler: async (item) => { await item.remove();  },
+    });
+  };
+
+  before(async () => {
+    WorkerState = workerState.setup({
+      table: helper.cfg.get('provisioner:workerStateTableName'),
+      credentials: helper.cfg.get('azure'),
+    });
+
+    let keyPrefix = helper.cfg.get('provisioner:awsKeyPrefix');
+    let pubKey = helper.cfg.get('provisioner:awsInstancePubkey');
+    let provisionerId = helper.cfg.get('provisioner:id');
+    let provisionerBaseUrl = helper.cfg.get('server:publicUrl') + '/v1';
+    WorkerType = workerType.setup({
+      table: helper.cfg.get('provisioner:workerTypeTableName'),
+      credentials: helper.cfg.get('azure'),
+      context: {keyPrefix, provisionerId, provisionerBaseUrl, pubKey},
+    });
+
+    await WorkerState.ensureTable();
+    await WorkerType.ensureTable();
+    await cleanTables();
+
+    wt = slugid.nice();
+  });
+
+  afterEach(async () => {
+    await cleanTables();
+  });
+
+  let addTestWorkerType = async () => {
+    await WorkerType.create(wt, {
+      launchSpecification: {},
+      minCapacity: 0,
+      maxCapacity: 100,
+      scalingRatio: 1.0,
+      minPrice: 0.0,
+      maxPrice: 1.0,
+      canUseOndemand: true,
+      canUseSpot: true,
+      instanceTypes: [
+        {instanceType: 'm1', capacity: 1},
+        {instanceType: 'm2', capacity: 2},
+      ],
+      regions: [],
+      lastModified: new Date(),
+      userData: {},
+      launchSpec: {},
+      secrets: {},
+      scopes: [],
+    });
+    await WorkerState.create({
```

Comment:
Instead of coding these into the test directly, can you please split them out like we have for worker type definitions?  it makes updating schemas and entities a whole lot easier.